app/services/queries/offices_queries.rb

```diff
@@ -0,0 +1,38 @@
+module Queries
+  class OfficesQueries
+    attr_reader :page, :records_per_page
+
+    def initialize(scope, page, records_per_page)
+      @scope = scope
+      @page = page
+      @records_per_page = records_per_page
+    end
+
+    def alphabetic
+      @scope
+        .includes(:ratings, photo_attachment: :blob)
+        .order(:name)
+        .limit(records_per_page)
+        .offset(page * records_per_page)
+    end
+
+    def averages(type)
+      @scope
+        .preload(:ratings, photo_attachment: :blob)
```

Comment:
`includes` switches dynamically from using two separate queries to creating a single LEFT OUTER JOIN to get the data. When running `@offices.geocoded.map`, a new where clause is added to the previous query. If a single query has been generated it will create a conflict.
In this case the use of `preload` is necessary to avoid issues, since it has the same function but always returns two separate queries.
Reference: https://www.bigbinary.com/blog/preload-vs-eager-load-vs-joins-vs-includes